scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -43,12 +68,23 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
               val pcoll = planner(opt).run(pipeline)
               beamMode.sink(sink) match {
                 case Some(ssink) =>
-                  ssink.write(pipeline, conf, pcoll)
+                  ssink.write(pcoll, conf)
                 case _ => throw new Exception(s"unknown sink: $sink when writing $pipe")
               }
               rec(xs)
             }
-            //TODO: handle Force and ToIterable
+            case OptimizedWrite(pipe, ToWrite.ToIterable(opt)) =>
+              val pcoll = planner(opt).run(pipeline)
+              val tempLocation = pcoll.getPipeline.getOptions.getTempLocation
+              require(tempLocation != null, "Temp location cannot be null when using toIterableExecution")
+
+              val outputPath = BeamWriter.addPaths(tempLocation, sourceCounter.getAndIncrement().toString)
+              // Here we add a sink transformation on the PCollection.
+              // This does not run till the final `pipeline.run` step
+              new BeamFileIO(outputPath).write(pcoll, conf)
+              iterablePipes += ((pipe, (pcoll.getCoder, outputPath)))
+
+            //TODO: handle Force
```

Comment:
force is almost the same as toIterable, you just materialize it and then make a PCollection to read it in the next pipeline, but you are smart to leave that to a follow up PR.

Reply:
Yup this will be done in a follow up patch